packages/pie-inline-choice/src/index.js

```diff
@@ -0,0 +1,41 @@
+import Main from "./main.jsx";
+import React from "react";
+import ReactDOM from "react-dom";
+
+export default class InlineChoice extends HTMLElement {
+
+  constructor() {
+    super();
+    this._model = null;
+    this._session = null;
+    this._rerender = () => {
+      if(this._model && this._session) {
+        let elem = React.createElement(Main, {
+          model : this._model,
+          session : this._session
```

Comment:
you'll want a callback here to pick up the session changes: `onSessionChanged: this.onSessionChanged`, in that method you'll need to dispatch an `new SessionChangedEvent`.